File: 2405. Optimal Partition of String/main.py
class Solution:
    def partitionString(self, s: str) -> int:
        count = [0] * 26
        num = 1

        for c in s:
            index = ord(c) - ord("a")

            if count[index] == 1:
                count = [0] * 26
                num += 1

            count[index] += 1

        return num


class Solution:
    def partitionString(self, s: str) -> int:
        mask = 0
        num = 1

        for c in s:
            bit = 1 << (ord(c) - ord("a"))

            if (mask & bit) != 0:
                mask = 0
                num += 1

            mask |= bit

        return num


class Solution:
    def partitionString(self, s: str) -> int:
        sets = set()
        num = 1

        for c in s:
            if c in sets:
                sets.clear()
                num += 1

            sets.add(c)

        return num


class Solution:
    def partitionString(self, s: str) -> int:
        maps = dict()
        num = 1

        for c in s:
            if c in maps:
                maps.clear()
                num += 1

            maps[c] = 1

        return num

File: 2405. Optimal Partition of String/main.rs
struct Solution;

impl Solution {
    pub fn partition_string(s: String) -> i32 {
        let mut count = vec![0; 26];
        let mut num = 1;

        for c in s.chars() {
            let index = c as usize - 'a' as usize;

            if count[index] == 1 {
                count = vec![0; 26];
                num += 1;
            }

            count[index] = 1;
        }

        num
    }
}

impl Solution {
    pub fn partition_string(s: String) -> i32 {
        let mut mask = 0;
        let mut num = 1;

        for c in s.chars() {
            let bit = 1 << (c as usize - 'a' as usize);

            if (mask & bit) != 0 {
                mask = 0;
                num += 1;
            }

            mask |= bit;
        }

        num
    }
}

use std::collections::HashSet;

impl Solution {
    pub fn partition_string(s: String) -> i32 {
        let mut sets = HashSet::new();
        let mut num = 1;

        for c in s.chars() {
            if sets.contains(&c) {
                sets.clear();
                num += 1;
            }

            sets.insert(c);
        }

        num
    }
}
